### PR TITLE
Nwfind fix

### PR DESCRIPTION
**Pull request by ddutt**
Network find had a bunch of fixes:

* Correct handling when both start and end times are provided or view=all
* Correct tagging of an address as bridged/routed
* Correct handling of an address that is an interface address on a device we're polling.

### CODE REVIEW

File: suzieq/engines/pandas/network.py [hunk 1]
```diff
@@ -179,7 +206,15 @@ def _find_addr_arp(self, addr: str, **kwargs) -> pd.DataFrame:
...
-                if 'vlan' not in ifdf.type.unique():
+                netaddr = ip_address(row.ipAddress)
+                # The code below checks if the provided address belongs to
+                # the interface subnet. If it does, its a bridged address
+                # else its a routed address i.e. you use routing to reach it.
+                if not ifdf.ipAddressList.apply(
+                        lambda x, addr:
+                        False if not x.any() else
+                        any(addr in ip_network(y, strict=False) for y in x),
+                        args=(netaddr,)).any():
```
Comment (claudiolor) [inline suzieq/engines/pandas/network.py:217]: This piece of code is not easy to follow, I would rename some of the variables:
```suggestion
                if not ifdf.ipAddressList.apply(
                        lambda addr_list, netaddr:
                        False if not addr_list.any() else
                        any(netaddr in ip_network(a, strict=False) for a in addr_list),
                        args=(netaddr,)).any():
```

Comment (ddutt) [reply suzieq/engines/pandas/network.py:217]: Fixed with diff var names

File: suzieq/engines/pandas/network.py [hunk 2]
```diff
@@ -75,14 +76,34 @@ def _find_address(self, addr: str, **kwargs) -> pd.DataFrame:
...
+                                  'state', 'type'],
+                         errors='ignore')
+            df['ipAddress'] = df.ipAddress.apply(
+                lambda x: ', '.join([y.split('/')[0] for y in x]))
+
+            df['bondMembers'] = ''
+            df['type'] = 'interface'
+            df['l2miss'] = False
+
+            return df[cols]
```
Comment (LucaNicosia) [inline suzieq/engines/pandas/network.py:105]: I think if one of the network columns is not in address, this line will fail

File: suzieq/engines/pandas/network.py [hunk 3]
```diff
@@ -179,7 +206,16 @@ def _find_addr_arp(self, addr: str, **kwargs) -> pd.DataFrame:
...
             if not ifdf.empty:
-                if 'vlan' not in ifdf.type.unique():
+                netaddr = ip_address(row.ipAddress)
+                # The code below checks if the provided address belongs to
+                # the interface subnet. If it does, its a bridged address
+                # else its a routed address i.e. you use routing to reach it.
+                if not ifdf.ipAddressList.apply(
+                        lambda subnets, netaddr:
+                        False if not subnets.any() else
+                        any(addr in ip_network(subnet, strict=False)
```
Comment (LucaNicosia) [inline suzieq/engines/pandas/network.py:216]: Both `ip_address(row.ipAddress)` and `ip_network(subnet, strict=False)` expect a `_RawIpAddress`. `ip_address` seems to work also with a string, but `ip_network` raises this exception:
```shell
luca> network find address='10.0.0.11'                                                                                                                                                                                                                                                                Error: 1
                                             error
0  ERROR: 'str' object has no attribute '_version'
```

**Review by LucaNicosia — CHANGES_REQUESTED** [review]
Some network tests are failing. The reason is the `ip_network` bug

File: suzieq/engines/pandas/network.py [hunk 4]
```diff
@@ -75,14 +76,34 @@ def _find_address(self, addr: str, **kwargs) -> pd.DataFrame:
...
+            # Is this a locally attached interface IP to a device we're polling
+            df = self._get_table_sqobj('address') \
+                .get(vrf=vrf, address=[addr], columns=['*'],
+                     **kwargs)
+            if df.empty:
+                return addr_df
+
+            df = df.drop(columns=['ipAddressList', 'ip6AddressList',
+                                  'state', 'type'],
+                         errors='ignore')
```
[the remark this answers is not in the document]
Comment (ddutt) [reply suzieq/engines/pandas/network.py:97]: Done

File: suzieq/engines/pandas/network.py (hunk 3 above)
Comment (ddutt) [reply suzieq/engines/pandas/network.py:216]: Fixed. Typo when I changed var names.

File: suzieq/engines/pandas/network.py (hunk 2 above)
Comment (ddutt) [reply suzieq/engines/pandas/network.py:105]: Fixed

**Pull request merged**